here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/ExisingMetadataProvider.kt

```diff
@@ -0,0 +1,93 @@
+package naksha.psql.executors.write
+
+import naksha.base.Int64
+import naksha.base.ListProxy
+import naksha.model.Metadata
+import naksha.model.TupleNumber
+import naksha.model.Version
+import naksha.model.request.WriteOp.WriteOp_C.DELETE
+import naksha.model.request.WriteOp.WriteOp_C.UPDATE
+import naksha.model.request.WriteOp.WriteOp_C.UPSERT
+import naksha.psql.PgCollection
+import naksha.psql.PgColumn
+import naksha.psql.PgSession
+import naksha.psql.PgUtil.PgUtilCompanion.quoteIdent
+import naksha.psql.executors.WriteExt
+import naksha.psql.executors.write.PgCursorUtil.collectAndClose
+
```

Comment:
I think this should be only temporary solution, because actually we have a tuple-cache, which as well holds metadata, so we should merge this with the general tuple cache. Maybe add a TODO comment, so we know that this function should be replaced with a `fetchTuple` call with `fetchMode` for metadata.

Reply:
Well it depends, my goal was to keep it local, and let garbage-collector to clear it.
If you want to keep whole DB in memory, then yes, we can use global long-living cache, but do we want to keep all history versions in cache?

Reply:
TODO added